File: benchmarks/tsb/bench_dt_millisecond_microsecond_nanosecond.ts
/**
 * Benchmark: dt_millisecond_microsecond_nanosecond — DatetimeAccessor millisecond, microsecond, nanosecond on 100k values
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const now = Date.now();
const data = Array.from({ length: ROWS }, (_, i) => new Date(now + i * 1_000));
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  s.dt.millisecond();
  s.dt.microsecond();
  s.dt.nanosecond();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.dt.millisecond();
  s.dt.microsecond();
  s.dt.nanosecond();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dt_millisecond_microsecond_nanosecond",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_dt_millisecond_microsecond_nanosecond.py
"""Benchmark: dt_millisecond_microsecond_nanosecond — pandas dt.microsecond, dt.nanosecond on 100k values"""
import json
import time
import pandas as pd
import numpy as np

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = pd.date_range("2020-01-01", periods=ROWS, freq="s")
s = pd.Series(data)

for _ in range(WARMUP):
    _ = s.dt.microsecond
    _ = s.dt.nanosecond

start = time.perf_counter()
for _ in range(ITERATIONS):
    _ = s.dt.microsecond
    _ = s.dt.nanosecond
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "dt_millisecond_microsecond_nanosecond", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
